api/service/prediction_job_service.go

```diff
@@ -42,6 +43,8 @@ type PredictionJobService interface {
 	ListContainers(env *models.Environment, model *models.Model, version *models.Version, predictionJob *models.PredictionJob) ([]*models.Container, error)
 	// StopPredictionJob deletes the spark application resource and cleans up the resource
 	StopPredictionJob(env *models.Environment, model *models.Model, version *models.Version, id models.ID) (*models.PredictionJob, error)
+	// ExecuteDeployment creates prediction job based on queue job that produce by CreatePredictionJob method
```

Comment:
I think `ExecuteDeployment` shouldn't be part of `PredictionJobService`.  
`PredictionJobService` should just submit the request to queue, and then queue+worker handle the rest.